tests/test.js

```diff
@@ -42,3 +42,28 @@ test('build next app with manifest and service worker', async () => {
   const str = await read(join(cwd, '.next/main.js'), 'utf8');
   expect(str).toEqual(expect.stringContaining('serviceWorker'));
 });
+
+test('pre-caching includes static assets and bundles', async () => {
+  const nextConf = forceProd(
+    withOffline(
+      withManifest({
+        manifest: {
+          name: 'next-app',
+        },
+      }),
+    ),
+  );
+
+  await nextBuild.default(cwd, nextConf);
+  const str = await read(join(cwd, '.next/service-worker.js'), 'utf8');
+
+  // Check that various bundles are getting entered into pre-cache manifest
```

Comment:
There are build hashes in the two assertions below that I avoided. I'm sure I could pipe the build ID into this test file, and will if you'd prefer that, but this is a good start.